### ISSUE TITLE
crushed fluid barrels release no pollution

### ISSUE THREAD

**Issue by Quezler** [Deadlock989/IndustrialRevolution#42]
saw this setup on the aforementioned server to get rid of excess fluids, thought about just crushing fluid barrels instead to save the conveyor overhead to simplify it, but they seemingly do not release pollution (text popup) in the same way as pipes/tanks/trains do 🤔 (just pointing it out, kindly keep the ability to crush barrels, but some pollution would be neat)

![Screen Shot 2021-04-22 at 15 04 59](https://user-images.githubusercontent.com/3179271/115719345-498a3a80-a37c-11eb-9811-53ab7b878fe8.png)

**Comment by Deadlock989** [comment]
This isn't really a bug or an issue, it's more of a feature request. I'm happy to fix obvious bugs or mistakes but the mod isn't being developed any further.

For what it's worth I did consider scrapping issues like this and even tried a couple of things out, as well as several other related issues and arguable "exploits" to do with pollution, but then had to start asking myself some more difficult questions about what the hell I was doing with my life.

**Comment by Deadlock989** [comment]
I went back and had another look at this. Scrapping machines can't produce a pop-up in the way that voiding fluid systems can - there are no runtime event hooks for recipe completion. However the scrapping machine does already produce some pollution when it works, so lubricant barrels, express belts etc. do already incur some pollution when scrapped - just not very much, and a lubricant barrel produces the same pollution as an empty barrel.

I realised the existing scrapping system could look for plastic and lubricant while it was searching for scrap values and add a further pollution penalty instead of adding "plastic scrap" to the outputs. This ends up as a multiplier, so it's particularly harsh with "large items" (items with a lot of raw materials in them already take longer to scrap, so they release more pollution per item). It is in no way "balanced", e.g. with other methods of getting rid of things, but it's something.

Also, I decided to return the flare stack from IR1 and allow it to burn lubricant as well, at a significant pollution penalty.

This will appear in IR 2.2.0 (likely end of June 2021).
